stdlib/public/stubs/Stubs.cpp

```diff
@@ -20,6 +20,12 @@
 #define _WITH_GETLINE
 #endif
 
+#if defined(__CYGWIN__)
+#include <stdio.h>
+typedef  void *locale_t;
+#define uselocale(x)  nullptr
```

Comment:
Setting the locale to C is an important part of the Swift's floating point parsing semantics, to ensure that numbers are parsed the same way regardless of the locale.  Please don't drop it.